gson/src/test/java/com/google/gson/functional/PrimitiveTest.java

```diff
@@ -280,86 +307,97 @@ public void testNumberDeserialization() {
     assertEquals(1L, actual.longValue());
   }
 
+  @Test
   public void testNumberAsStringDeserialization() {
     Number value = gson.fromJson("\"18\"", Number.class);
     assertEquals(18, value.intValue());
   }
 
+  @Test
   public void testPrimitiveDoubleAutoboxedSerialization() {
     assertEquals("-122.08234335", gson.toJson(-122.08234335D));
     assertEquals("122.08112002", gson.toJson(122.08112002D));
   }
 
+  @Test
   public void testPrimitiveDoubleAutoboxedDeserialization() {
     double actual = gson.fromJson("-122.08858585", double.class);
-    assertEquals(-122.08858585D, actual);
+    assertEquals(-122.08858585D, actual, 0);
 
     actual = gson.fromJson("122.023900008000", Double.class);
-    assertEquals(122.023900008D, actual);
+    assertEquals(122.023900008D, actual, 0);
   }
 
+  @Test
   public void testPrimitiveDoubleAutoboxedInASingleElementArraySerialization() {
     double[] target = {-122.08D};
     assertEquals("[-122.08]", gson.toJson(target));
     assertEquals("[-122.08]", gson.toJson(target, double[].class));
     assertEquals("[-122.08]", gson.toJson(target, Double[].class));
   }
 
+  @Test
   public void testDoubleAsStringRepresentationDeserialization() {
     String doubleValue = "1.0043E+5";
     Double expected = Double.valueOf(doubleValue);
     Double actual = gson.fromJson(doubleValue, Double.class);
     assertEquals(expected, actual);
 
     double actual1 = gson.fromJson(doubleValue, double.class);
-    assertEquals(expected.doubleValue(), actual1);
+    assertEquals(expected, actual1, 0);
   }
 
+  @Test
   public void testDoubleNoFractAsStringRepresentationDeserialization() {
     String doubleValue = "1E+5";
     Double expected = Double.valueOf(doubleValue);
     Double actual = gson.fromJson(doubleValue, Double.class);
     assertEquals(expected, actual);
 
     double actual1 = gson.fromJson(doubleValue, double.class);
-    assertEquals(expected.doubleValue(), actual1);
+    assertEquals(expected, actual1, 0);
   }
 
+  @Test
   public void testDoubleArrayDeserialization() {
       String json = "[0.0, 0.004761904761904762, 3.4013606962703525E-4, 7.936508173034305E-4,"
               + "0.0011904761904761906, 0.0]";
       double[] values = gson.fromJson(json, double[].class);
-      assertEquals(6, values.length);
-      assertEquals(0.0, values[0]);
-      assertEquals(0.004761904761904762, values[1]);
-      assertEquals(3.4013606962703525E-4, values[2]);
-      assertEquals(7.936508173034305E-4, values[3]);
-      assertEquals(0.0011904761904761906, values[4]);
-      assertEquals(0.0, values[5]);
+      assertEquals(6, values.length, 0);
```

Comment:
This looks incorrect / unnecessary since the values here are `int` so there is no need to use `assertEquals(double, double, double)`:
```suggestion
      assertEquals(6, values.length);
```

In case you do want to change this, maybe the following should also be changed to use `assertEquals(float, float, float)` instead of the `double` variant by appending`F` to `0.00001`:
https://github.com/google/gson/blob/1a2170b99c9d293e825bf8f511b191196326ea03/gson/src/test/java/com/google/gson/JsonPrimitiveTest.java#L100
(though this is not an issue you introduced)

Reply:
Ohh yes, good point. 😄 
If we don’t change all the assertions to Truth, the next PR will be "simplify assertions". I’ll fix it in that PR.